spock-core/src/main/java/org/spockframework/compiler/DeepBlockRewriter.java

```diff
@@ -61,13 +62,55 @@ public void visitAssertStatement(AssertStatement stat) {
 
   @Override
   protected void doVisitExpressionStatement(ExpressionStatement stat) {
-    super.doVisitExpressionStatement(stat);
+    InteractionRewriter rewriter = visitInteractionAwareExpressionStatement(stat);
 
     boolean handled = (stat == lastSpecialMethodCallStat && !(currSpecialMethodCall.isWithCall() || currSpecialMethodCall.isGroupConditionBlock())) // don't process further
-        || handleInteraction(stat)
+        || handleInteraction(rewriter, stat)
         || handleImplicitCondition(stat);
   }
 
+  private InteractionRewriter visitInteractionAwareExpressionStatement(ExpressionStatement stat) {
+    InteractionRewriter rewriter = new InteractionRewriter(resources, getCurrentWithOrMockClosure());
+    if(isInterationExpression(rewriter, stat)) {
```

Comment:
nit: missing space (also for other `if` statements in `doVisitBinaryExpression`)